### PR TITLE
Mining Update I [WIP | DNM]

### PR DESCRIPTION
**Pull request by ghost**
Main changes listed below.

**Bait** can be used to distract hostile simple animals in order to escape an attack. Simply throw the bait - if a monster is hit, it will lose interest in you and eat the bait instead.

**Mining Charges** are explosive mining tools that can be bought from mining vendors. They have a decent explosion radius and are much more efficient than gibtonite. Mining charges, unless emagged, can only be placed on rock and minerals.

**Capture Spheres** are used to capture simple animals for companionship. Simply throw the sphere at a simple animal. Depending on its health, it may break out. However, if it doesn't, you can rename the mob and release it at will by throwing the ball. You can suck it back into the capture sphere by activating it in your hand next to the monster - this will also revive it. Now you can have your own backpack minions!

**Portable Refrigerators** are the solution to all of your hivelord-related problems. Tired of those ever-so-useful hivelord cores decaying and becoming useless? No more with the FreezeWare Cryogenic Refrigerator (patent pending)! Simply place a hivelord core into the refrigerator and it will indefinitely be preserved until it is needed. _NOTICE: This device may only hold one hivelord core at a come. In addition, tampering with the device may cause unexpected side effects. Do not place a hivelord core into a refrigerator that has been tampered with._

**Mythril Ore** is a newly-discovered ore on the mining asteroid. It's about as rare as diamond and had a unique trait: it is very malleable at any temperatures until shapen, at which point it is very sturdy and hard to break - about as strong as plasteel. One can shape some tools with their bare hands using this metal, in addition to creating armor plating which can be used in place of goliath plating on hardsuits (but only hardsuits, no mechs or boxes). It can also be used to plate drills to make them more effective.

**Will o' the Wisps** are new mining monsters. These shimmering globules of light will hypnotize their target into a trance, drawing them through tunnels. They are very fragile but hit quite hard in an emergency - however, they constantly run away from the player, similar to a goldgrub. Upon death, their energies will dissipate, leaving behind an energetic sphere. This sphere's primary function is to recharge energy-based items. It can be used on APCs, SMES, and energy-based weaponry to recharge them completely, and has the secondary function of being fairly robust.

And some miscellaneous changes.
- Pickaxes now have a description.
- Some unused files have been removed or repurposed from mining.dmi.
- Stimpack kits no longer come with seven epinephrine medipens.
- Ore boxes drop some of their ore after an explosion, and plating them with goliath hide makes them blast-proof.

### CODE REVIEW

**Comment by RemieRichards** [comment]
You're going to need to exclude certain mobs from the capture ball (for example, Drones are simple animals too)

**Comment by WJohn** [comment]
MINING CHARGES.

YES!

FINALLY!

**Comment by JJRcop** [comment]
pokemon in spaaaaaaaaaaaace

**Comment by Incoming5643** [comment]
Having the game force speech lines on the player always struck me as really gimmicky, but cool beyond that.

Both relived and disappointed that the balls do a client check on simple animals, capturing a ling headslug could've been hilarious.

**Comment by AlexCrimson** [comment]
Whats happens if you throw bait then start hitting the mob? I assume it starts attacking you again? Even so, could you not throw bait then drop a gibonite ore/mining charge near it and blow it up whilst its distracted?

Pokeballs sound neat. The mining vendor really needs some funny items that are not completely useless.

**Comment by PKPenguin321** [comment]
Pokemon in space is totally radical
(make an admin variant or something that can capture monkeys so we can have illegal underground monkey-cockfights)

Mining getting infinity bombs sounds cool

Bait is a good and well rounded idea that I can see myself using.

**Comment by SconesC** [comment]
bait :+1: 
charges :+1: 

fuck pokemon, it may be cool but muh immersion

also >capturing goliaths to spawn later

**Comment by vista-** [comment]
can you make bait work on people?

File: code/modules/mining/bugBait.dm [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//BAIT: Stops a mining creature from following you, distracting it if it hits
+/obj/item/weapon/miningBait
```
Comment (Iamgoofball) [inline code/modules/mining/bugBait.dm]: change to mining_bait for standardization

Comment (Iamgoofball) [inline code/modules/mining/bugBait.dm]: Same w/ the rest of the objects

**Comment by Iamgoofball** [comment]
THE SPARKS ARE FLYING IN THIS HEATED BATTLE

**Comment by Miauw62** [comment]
GOTTA DUNK EM ALL
Op 28-mei-2015 07:19 schreef "Iamgoofball" <notifications@github.com>:

**Comment by Incoming5643** [comment]
is there any situation at all where the capture sphere is worse than the Lazarus injector

**Comment by Fox-McCloud** [comment]
:+1:  To this all; miners don't get enough appreciation; this is a pretty damn cool update.

**Comment by optimumtact** [comment]
It's not pokemon you dicks, it's backpack monsters

**Comment by Fox-McCloud** [comment]
One suggestion I have---it would have to be snowflaked out, but....

Make it so that mobs loyal to you (ie: one you've re-animated) are caught without fail.

**Comment by pudl** [comment]
:-1: to mining charges being better than gibtonite. why ever mine gibtonite if you can get superior mining charges? gibtonite is risky to obtain and very volatile, it should be stronger

**Comment by Jalleo** [comment]
Why not make it so the mining charge uses gibonite?

**Comment by SconesC** [comment]
How big is "much more effective than gibonite"? There are varying degrees of it.

**Comment by bgobandit** [comment]
More pertinently, how big is it compared to the syndicate bomb out of an uplink?

**Comment by Fox-McCloud** [comment]
Suggestion: Emagged capture balls set the mob's faction to you and only you.

Related: mobs with the same faction as you can be captured 100% regardless of health and do not get their faction over-ridden.

File: code/modules/mining/miningCharge.dm [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	var/detonating = 0 //If the charge is currently primed
+	var/safety = 1 //If the charge can be put on things other than rocks
+	var/explosionPower = 2 //The power of the explosion; larger powers = bigger boom
+	var/atom/movable/putOn = null //The atom the charge is on
+	var/primedOverlay = null
+
+/obj/item/device/miningCharge/emag_act(mob/user)
+	if(!safety)
+		return
+	user << "<span class='warning'>You press the emag onto [src], disabling its safeties.</span>"
```
Comment (Iamgoofball) [inline code/modules/mining/miningCharge.dm]: use cryptographic sequencer for situations like these, we use emag as shorthand, but game text is more "professional" in a way and uses the full name.

File: code/modules/mining/miningCharge.dm [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	safety = 0
+	explosionPower-- //Makes them less powerful when you emag them - for balance
```
Comment (Iamgoofball) [inline code/modules/mining/miningCharge.dm]: This is a little iffy. Emagging something making it less useful is rather annoying and I'd prefer if this was removed and changed to have a more obvious sprite when emagged.

**Comment by Xhuis** [comment]
Explosions are not as effective at devastation but are better at breaking rock. Will respond more when I get home.

File: code/modules/mining/bugBait.dm (hunk 1 above)
Comment (Xhuis) [inline code/modules/mining/bugBait.dm]: no

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 4]
```diff
@@ -263,21 +263,37 @@
...
-		desc = "The remains of a hivelord that have become useless, having been left alone too long after being harvested."
+	..()
+	SSobj.processing += src
+
+/obj/item/asteroid/hivelord_core/Destroy()
+	SSobj.processing -= src
+	..()
+
+/obj/item/asteroid/hivelord_core/process()
+	spawn(10)
```
Comment (Iamgoofball) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: the spawn isn't needed

File: code/modules/mining/miningCharge.dm (hunk 2 above)
Comment (Dorsisdwarf) [inline code/modules/mining/miningCharge.dm]: Yeah - emag is game slang, not the actual term

File: code/modules/mining/captureSphere.dm [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	return
+
+/obj/item/device/captureSphere/proc/Release(var/mob/user)
+	if(!capturedMob)
+		return
+	if(capturedMob && capturedMob.loc != src)
+		return
+	user.say("[pick("Go", "Get 'em")], [capturedMob.name]!")
+	capturedMob.loc = get_turf(src)
+	capturedMob.say("[capturedMob.name]!")
```
Comment (Dorsisdwarf) [inline code/modules/mining/captureSphere.dm]: Not even being subtle here

**Comment by Dorsisdwarf** [comment]
The capture ball issue Remie mentioned is still vital!

Many things are simple mobs, such as slimes, mining bots, ascendant shadowlings, syndicate gunners, space clowns, space pirates, space russians, and drones

**Comment by Miauw62** [comment]
capturing an ascendant shadowling in one of these balls would be fucking hilarious though

you could even add having the chacne for capture be <100% if their health is over 50%

**Comment by duncathan** [comment]
Ascendant shadowing is a legendary

**Comment by Iamgoofball** [comment]
Just allow capturing all simple mobs so that I can pit fight Ian VS Runtime, "muh reels" be damned.

**Comment by AlexCrimson** [comment]
Oh shit can we get simple mob evolutions? Ian evolving into Deathsquad Ian would be amazing.

**Comment by Xhuis** [comment]
Capturing all simple mobs is currently allowed. I will blacklist abusable mobs (like clowns and syndicates) but otherwise catching them all is the whole idea. The capturing has sanity checks, so drones and stuff can't be captured.

**Comment by Iamgoofball** [comment]
What's abusable about capturing Syndicates and Clowns? The NPC editions, that is. I see no problem with that. Honestly if you can get their health down low enough to safely capture them then I'd say you deserve it.

File: code/modules/mining/bugBait.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/mining/bugBait.dm]: why

**Comment by AlexCrimson** [comment]
The Syndie mobs have a pretty mean weapon, usually.

**Comment by Xhuis** [comment]
@Iamgoofball: Imagine killing a syndie, looting its weapon, then reviving it, killing it again, looting its weapon, reviving it...

**Comment by Incoming5643** [comment]
once a humanoid simple animal dies it becomes a carbon corpse, unless the ball remakes a new mob on heal it ceases to exist utterly.

Hell you could even debrain a dead russian, put a player brain in there, and create a carbon player russian.

**Comment by Kromgar** [comment]
Gotta catch em all...

File: code/modules/mining/bugBait.dm (hunk 1 above)
Comment (Xhuis) [inline code/modules/mining/bugBait.dm]: cuz

**Comment by PKPenguin321** [comment]
ohohoho
adamantine tools are for breaking out of the gulag i see

**Comment by Xhuis** [comment]
;)

File: code/modules/mining/captureSphere.dm (hunk 5 above)
Comment (Iamgoofball) [inline code/modules/mining/captureSphere.dm]: i see no problem with this

**Comment by Kromgar** [comment]
>Replaces adamantine with Mythril

Come on Xhuis how will i release the circus now?

**Comment by Xhuis** [comment]
I didn't want to replace it, but there was one problem: adamantine golems. I'll probably add adamantine as a separate green metal though, it's not gone entirely.

**Comment by Xhuis** [comment]
Will do this later when I have less to work on.

**Comment by WJohn** [comment]
RIP

**Pull request closed without merging**